epson-workforce-635-nx625-series: init at 1.0.1

jorsn opened the pull request:
###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ]  Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after) \
    ***`nix path-info -S nixpkgs.epson-workforce-635-nx625-series` or what? Before and after what?***
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### More

Tested with my Epson Stylus Office BX525WD wireless inkjet printer with success.

jorsn committed:
epson-workforce-635-nx625-series: init at 1.0.1

jorsn committed:
epson-workforce-635-nx625-series: added to all-packages.nix

jorsn committed:
epson-workforce-635-nx625-series: fixed build errors

variables `maintainers` and `licenses` were invisible, leading to build
errors; fixed this.

jorsn commented:
From GrahamcOfBorg-eval log:
> access to URI 'https://download.ebz.epson.net/dsc/op/stable/SRPMS/epson-inkjet-printer-workforce-635-nx625-series-1.0.1-1lsb3.2.src.rpm' is forbidden in restricted mode

I have no idea what to do about this.

Mic92 commented on pkgs/misc/drivers/epson-workforce-635-nx625-series/default.nix:
```diff
@@ -0,0 +1,98 @@
+  name = "epson-inkjet-printer-workforce-635-nx625-series";
+  version = "1.0.1";
+
+  src = builtins.fetchurl {
```
Use `fetchurl` instead of `builtins.fetchurl` to avoid the evaluation warning.

jorsn committed:
epson-workforce-635-nx625-series: use lib.fetchurl

builtins.fetchurl (was used) cannot access uris not in `allowed-uris`
when built in restricted mode (like ofborg does)

Mic92 commented:
@GrahamcOfBorg build epson-workforce-635-nx625-series

GrahamcOfBorg commented:
<!--REQUEST_ID=74144bdb-99ee-4ec0-8418-6930b25594b7-->
No attempt on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.48752&attempt_id=22ffdf1f-b73d-49c1-8804-216de03a881b)

The following builds were skipped because they don't evaluate on x86_64-darwin: epson-workforce-635-nx625-series

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnfree = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnfree = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=74144bdb-99ee-4ec0-8418-6930b25594b7-->
No attempt on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.48752&attempt_id=195b88e5-cf61-4957-8b56-d3301c8e2a85)

The following builds were skipped because they don't evaluate on x86_64-linux: epson-workforce-635-nx625-series

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnfree = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnfree = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=74144bdb-99ee-4ec0-8418-6930b25594b7-->
No attempt on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.48752&attempt_id=2d84b2c1-b887-4174-815d-8bb202207356)

The following builds were skipped because they don't evaluate on aarch64-linux: epson-workforce-635-nx625-series

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowUnfree = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowUnfree = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

jorsn commented on pkgs/misc/drivers/epson-workforce-635-nx625-series/default.nix:
```diff
@@ -0,0 +1,98 @@
+        };
+    '';
+    downloadPage = https://download.ebz.epson.net/dsc/du/02/DriverDownloadInfo.do?LG2=EN&CN2=&DSCMI=16857&DSCCHK=4334d3487503d7f916ccf5d58071b05b7687294f;
+    license = with lib.licenses; [ lgpl21 epson ];
```
The package includes lgpl source code and proprietary object files which are linked against. Since the proprietary and lgpl parts are distributed in one rpm I don't think one may leave out the epson license even if splitting the package into two subpackages in nixpkgs.

jorsn replied:
The package includes lgpl source code and proprietary object files which are (dynamically) linked against. Since the proprietary and lgpl parts are distributed in one rpm I don't think one may leave out the epson license even if splitting the package into two subpackages in nixpkgs.

jorsn commented:
Is there any problem why this doesn't get merged? After all, this is without any comments or failed tests since Oct 22.

Mic92 commented:
@jorsn Thanks for the update. This pull request simply went under.
